apps/arena/lib/arena/bots/pathfinding_grid.ex

```diff
@@ -0,0 +1,101 @@
+defmodule Arena.Bots.PathfindingGrid do
+  @moduledoc """
+  Generates a grid for current map.
+  """
+  alias Arena.Configuration
+  use GenServer
+  @update_interval_ms 3_600_000
+  require Logger
+
+  def start_link(_opts) do
+    GenServer.start_link(__MODULE__, [], name: __MODULE__)
+  end
+
+  def init(_opts) do
+    if System.get_env("PATHFINDING_TEST") == "true" do
+      Process.send_after(self(), :update_config, 1_000)
+    end
+
+    {:ok, %{}}
+  end
+
+  def get_map_collision_grid(map_name, from_pid) do
+    GenServer.cast(__MODULE__, {:get_map_collision_grid, map_name, from_pid})
+  end
+
+  def handle_info(:update_config, _state) do
+    state =
+      Configuration.get_current_maps_configuration()
+      |> Enum.reduce(%{maps: []}, fn map, acc ->
+        map_obstacles =
+          Enum.reduce(map.obstacles, {[], 1}, fn obstacle, {obstacles_acc, current_id} ->
+            # The following is only done so Rust doesn't complain about Entity type param
+            updated_obstacle =
+              obstacle
+              |> Map.take([
+                :id,
+                :shape,
+                :position,
+                :radius,
+                :vertices,
+                :speed,
+                :category,
+                :direction,
+                :is_moving,
+                :name
+              ])
+              |> Map.put(:id, current_id)
+              |> Map.put(:shape, get_shape(obstacle.shape))
+              |> Map.put(:category, :obstacle)
+              |> Map.put(:is_moving, false)
+              |> Map.put(:direction, %{x: 0.0, y: 0.0})
+              |> Map.put(:speed, 0.0)
+
+            {obstacles_acc ++ [updated_obstacle], current_id + 1}
+          end)
+          |> elem(0)
+          |> Enum.map(fn obstacle -> {obstacle.id, obstacle} end)
+          |> Map.new()
+
+        collision_grid =
+          case AStarNative.build_collision_grid(map_obstacles) do
+            {:ok, collision_grid} ->
+              collision_grid
+
+            {:error, reason} ->
+              Logger.error("Grid construction failed with reason: #{inspect(reason)}")
+
+              %{}
+          end
```

Comment:
can we return nil on the error case? I don't see too much sense in having to handle an empty map besides the binary and nil cases